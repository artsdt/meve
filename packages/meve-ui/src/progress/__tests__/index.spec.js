import Progress from '..'
import { mount } from '@vue/test-utils'

test('test progress with value', () => {
  const wrapper = mount(Progress, {
    propsData: {
      value: 50,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test progress custom style', () => {
  const wrapper = mount(Progress, {
    propsData: {
      value: 50,
      trackSize: 2,
      trackColor: 'red',
      color: 'green',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test progress show label', () => {
  const wrapper = mount(Progress, {
    propsData: {
      value: 50,
      label: true,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test progress default slot', () => {
  const wrapper = mount(Progress, {
    propsData: {
      value: 50,
      label: true,
    },
    scopedSlots: {
      default: '<div>default slot</div>',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})
