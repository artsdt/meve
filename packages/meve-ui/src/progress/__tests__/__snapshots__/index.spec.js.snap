// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test progress custom style 1`] = `
"<div class=\\"m-progress\\">
  <div class=\\"m-progress__track\\" style=\\"background: red; height: 2px;\\">
    <div class=\\"m-progress__bar\\" style=\\"background: green; width: 50%;\\"></div>
  </div>
</div>"
`;

exports[`test progress default slot 1`] = `
"<div class=\\"m-progress\\">
  <div class=\\"m-progress__label\\">
    <div>default slot</div>
  </div>
  <div class=\\"m-progress__track\\">
    <div class=\\"m-progress__bar\\" style=\\"width: 50%;\\"></div>
  </div>
</div>"
`;

exports[`test progress show label 1`] = `
"<div class=\\"m-progress\\">
  <div class=\\"m-progress__label\\">50%</div>
  <div class=\\"m-progress__track\\">
    <div class=\\"m-progress__bar\\" style=\\"width: 50%;\\"></div>
  </div>
</div>"
`;

exports[`test progress with value 1`] = `
"<div class=\\"m-progress\\">
  <div class=\\"m-progress__track\\">
    <div class=\\"m-progress__bar\\" style=\\"width: 50%;\\"></div>
  </div>
</div>"
`;
